addons/purchase_stock/models/product.py

```diff
@@ -126,6 +127,8 @@ def action_set_supplier(self):
         orderpoint = self.env['stock.warehouse.orderpoint'].browse(orderpoint_id)
         if not orderpoint:
             return
+        if self.partner_id.purchase_warn == 'block':
+            raise UserError(_('This vendor cannot be set as a supplier: %s', self.partner_id.purchase_warn_msg))
```

Comment:
If you are adding text, the same text must added in alphabetical order in the corresponding .pot file.
Alphabetical order is based on the body of the text, here it's `'This vendor cannot be set as a supplier: %s'`